drivers/usb/device/usb_dc_stm32.c

```diff
@@ -990,10 +989,17 @@ void HAL_PCDEx_SetConnectionState(PCD_HandleTypeDef *hpcd, uint8_t state)
 
 	usb_disconnect = device_get_binding(
 				DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_CONTROLLER);
-
 	gpio_pin_configure(usb_disconnect,
-			   DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_PIN,
-			   DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_FLAGS |
-			   (state ? GPIO_OUTPUT_ACTIVE : GPIO_OUTPUT_INACTIVE));
+			   DT_INST_0_ST_STM32_USB_DISCONNECT_GPIOS_PIN, GPIO_DIR_OUT);
+
+	if (state) {
+		gpio_pin_write(usb_disconnect,
```

Comment:
This function is deprecated. See the mailing list for the GPIO announcement, this needs to use the new GPIO API.

Reply:
I only changed "usb_dc_detach". The changes shown at line 993 are only marked because the file got  a bit shorter. ;)

Reply:
@StefJar new GPIO API was merged recently. Your patch changes to gpio_ping_configure() are wrong/unnecessary, probably failed rebase/conflict resolution.

Reply:
hey ho
merged  a latest zephyr master branch and re-commited that changes. Added a extra commend. So it should be fine for you guys